src/App.js

```diff
@@ -121,40 +103,48 @@ class App extends Component {
 
   render() {
     return (
-      <main className="App">
-      {this.state.user ?
-        <Header 
-          buttonDisplay={this.logOut} 
-          user={this.state.user}
-          buttonText='Log Out'
-          title={`Welcome ${this.state.user.name}`}
-          /> :
-        <Header 
-          buttonDisplay={this.showLoginPage}
-          buttonText='Log In'
-          title='Welcome to Rancid Tomatillos'
-          />
-      }
-        {this.state.login &&
-          <Login
-            fetchUserData={this.fetchUserData} 
-            closeLoginPage={this.closeLoginPage}/>}
-        {this.state.homepage &&
-          <section className="home-page">
-            <CardContainer 
-              movies={this.state.movies} renderSpecificMovie={this.renderSpecificMovie}/>
-          </section>
-        }
-        {this.state.selectedMovie &&
-            <Movie 
-              movie={this.state.selectedMovie}
-              closeMovieDetail={this.closeMovieDetail}
+      <BrowserRouter>
+        <main className="App">
+        {this.state.user ?
+          <Header 
+            buttonDisplay={this.logOut} 
+            user={this.state.user}
+            buttonText='Log Out'
+            title={`Welcome ${this.state.user.name}`}
+            /> :
+          <Header 
+            buttonDisplay={this.showLoginPage}
+            buttonText='Log In'
+            title='Welcome to Rancid Tomatillos'
             />
         }
-        {this.state.error &&
-          <ErrorPage errorCode={this.state.error}/>
-        }
-      </main>
+          <Switch>
+            <Route path="/" exact render={() => 
+              <CardContainer
+                movies={this.state.movies} 
+                renderSpecificMovie={this.renderSpecificMovie} 
+              />
+              } 
+            />
+            <Route path="/login" exact render={() => 
+              <Login
+                fetchUserData={this.fetchUserData}
+              />
+            } 
+          />
+            <Route path="/movies/:id" render={() => 
+              <Movie 
+                movie={this.state.selectedMovie}
+              />
+              } 
+            />
+            <Route path="/error" exact render={() => <ErrorPage />} />
+          </Switch>
+          {this.state.error &&
+            <ErrorPage errorCode={this.state.error}/>
+          }
+        </main>
+      </BrowserRouter>
     );
```

Comment:
Nice implementation of React Router! It's a whole lot easier than manipulating displays, right??

Reply:
I love it, just started diving in. Excited to learn more about react router